power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -350,6 +354,79 @@ template <bool sym> class YBus {
         admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
     }
 
+    template <UpdateIsProgressive T>
+    void update_admittance_increment(std::shared_ptr<T const> const& math_model_param_incrmt,
+                                     bool is_decrement = false) {
+        const double mode = is_decrement ? -1.0 : 1.0;
+        if (!is_decrement) {
+            // overwrite the old cached parameters in increment mode (update)
+            math_model_param_incrmt_ = math_model_param_incrmt;
+        } else if (!decremented_) {
+            math_model_param_incrmt_ = {}; // reset recorded increment that is already decremented
+            return;
+        }
+
+        // construct admittance data
+        ComplexTensorVector<sym> admittance(nnz());
+        assert(Idx(admittance_->size()) == nnz());
+        std::ranges::copy(admittance_->begin(), admittance_->end(), admittance.begin());
+        auto const& b_param = math_model_param_incrmt_->branch_param;
+        auto const& b_param_2c = math_model_param_incrmt_->branch_param_to_change;
+        auto const& s_param = math_model_param_incrmt_->shunt_param;
+        auto const& s_param_2c = math_model_param_incrmt_->shunt_param_to_change;
+        auto const& y_bus_element = y_bus_struct_->y_bus_element;
+        auto const& y_bus_entry_indptr = y_bus_struct_->y_bus_entry_indptr;
+        auto const& y_bus_pos_in_entries = y_bus_struct_->y_bus_pos_in_entries;
```

Comment:
do you really use these so often that you need abbreviated aliases for them?

Reply:
I believe once is enough for the pain xD. I'll see about the first 4